### DISCUSSION TITLE
Email enumeration in password reset

### DISCUSSION THREAD

**Discussion by mirko-kienle** [statamic/cms#5240] [Q&A] [answered]
Hi, we've just finished building our first Statamic site for a client. The client has now had a security assessment run by an external infosec firm and one of their findings was the following:

> **Description:** User enumeration is when an attacker can determine if a username presented to the application is valid or not, due to the message returned from the server in response to authentication or password reset type of facility. This would allow the attacker to gain a list of valid usernames which could then be utilised in a brute-force attack or targeted phishing.

> **Details:** During the assessment a method of username enumeration was identified. Using the forgotten password function, the system would identify if the account entered was valid upon submission responding to the user with a non- generic message. If the account provided was valid then the system would respond that the email had been successfully sent; if the account was not valid then the user would receive a notification that a user could not be found with an email address. Given that no generic message is provided when the function is used, and that during the testing no rate limiting was identified for this; a malicious user can automate these requests to identify a list of valid usernames.

<img width="224" alt="Screenshot 2022-02-09 at 10 43 52" src="https://user-images.githubusercontent.com/39773181/153170232-a923af51-a016-436c-8829-59cc79d35d7a.png">

<img width="272" alt="Screenshot 2022-02-09 at 10 43 57" src="https://user-images.githubusercontent.com/39773181/153170262-d2261e61-bbec-4c0c-880f-b02cfa95e8c7.png">

This concerns the password reset functionality at `/cp/auth/password/reset`.

I've seen that you explicitly mention "Username or email address enumeration" as non-qualifying vulnerabilities in your security policy.

I guess my question would be why? It is a clear attack vector, even if the possible uses are limited. Best practice would seem to be serving generic messages instead of providing feedback on the (in)validity of submitted email addresses. Any guidance would be appreciated, thanks! I'm happy to open a corresponding issue if that's the best course of action.

**Comment by edalzell** [comment, marked as answer]
Sounds like a good feature request, I'd open it on the `ideas` repo.

**Reply by mirko-kienle** [reply]
Thanks for pointing me in the right direction @edalzell, appreciate it

**Reply by mirko-kienle** [reply]
https://github.com/statamic/ideas/issues/752
